platform/platform-impl/src/com/intellij/toolWindow/InternalDecoratorImpl.kt

```diff
@@ -593,10 +593,11 @@ class InternalDecoratorImpl internal constructor(
         ToolWindowAnchor.LEFT -> !isSplitter || isVerticalSplitter || !isFirstInSplitter
         else -> false
       }
-      val top = if (isSplitter && (anchor == ToolWindowAnchor.RIGHT || anchor == ToolWindowAnchor.LEFT) && windowInfo.isSplit && isVerticalSplitter) -1 else 0
+      val top = if (toolWindow.type != ToolWindowType.FLOATING && (!isVerticalSplitter || isFirstInSplitter) &&
```

Comment:
I don't understand what's going on here yet. The issue is about left/right borders, but these changes only affect the top border.

Reply:
The change in line 573 affects all borders, but I'm reworking this change to fix remaining issues. Stay tuned.